### PR TITLE
Add a new document type "Vehicle Recalls and Faults alert"

### PR DESCRIPTION
**Pull request by tadast**
This new document type is an attempt by @keithiopia and me to replace the archaic vehicle recalls tool http://www.dft.gov.uk/vosa/apps/recalls, as suggested by someone in the organisations that owns the current tool. 

This is just a MVP to see how well finders are suited to solve this problem. We would ideally need to add more manufacturers, potentially grouped by "Item type" (not sure what's the best way to implement it though)

I also needed to update `gds-sso` gem, because it depends on a newer version of `govuk_content_models`, which needs to be updated due to a new artefact type.

Ideally we'd like to test this in a preview environment, but since the data gets wiped every night it is not ideal. Any suggestions how best to try it out?

It's my first PR on specialist-publisher, so please review carefully :)

## Todo before merging

- [x] Release a new version of govuk_content_models gem with [this change](https://github.com/alphagov/govuk_content_models/pull/300)
- [x] update govuk_content_models version in panopticon
- [x] update govuk_content_models version in specialist-publisher (this PR?)
- [x] add new document type schema to rummager

### CODE REVIEW

**Comment by keithiopia** [comment]
We can get around that preview/testing issue by creating drafts in the production environment then publishing in preview.

File: README.md [hunk 1]
```diff
@@ -49,19 +49,19 @@ $ bundle exec rake
...
 6. Define the factory with the builder in `app/lib/specialist_publisher_wiring.rb`.
 7. Define the validatable document factory in `app/models/document_factory_registry.rb`
 8. Define a repository in `app/repositories/repository_registry.rb`
-9. Add observers, along with formatters required:
-  - `document_type_publication_alert_formatter.rb` in `app/exporters/formatters/`
-  - `document_type_artefact_formatter.rb` in `app/lib/` for Panopticon
-  - `document_type_indexable_formatter.rb` app/lib/` for Rummager
-  - define a factory for `document_type_panopticon_registerer`, `document_type_rummager_indexer`, `document_type_rummager_deleter` and `document_type_content_api_exporter` in `app/lib/specialist_publisher_wiring.rb`
-  - add an Observers registry for the document type and add it to the hash in the `observers_registry` method in `app/lib/specialist_publisher`
+9. Add observers, along with formatters required. In in `app/exporters/formatters/`:
```
Comment (tommyp) [inline README.md]: Double `In in` here

File: app/lib/specialist_publisher_wiring.rb [hunk 2]
```diff
@@ -91,7 +91,7 @@
 
   define_factory(:countryside_stewardship_grant_builder) {
     SpecialistDocumentBuilder.new("countryside_stewardship_grant",
-    get(:validatable_document_factories).countryside_stewardship_grant_factory)
+      get(:validatable_document_factories).countryside_stewardship_grant_factory)
```
Comment (tommyp) [inline app/lib/specialist_publisher_wiring.rb]: If you're indenting this, can you drop the closing bracket onto the next line?

File: app/lib/specialist_publisher_wiring.rb [hunk 3]
```diff
@@ -124,6 +124,11 @@
       get(:validatable_document_factories).raib_report_factory)
   }
 
+  define_factory(:vehicle_recalls_and_faults_alert_builder) {
+    SpecialistDocumentBuilder.new("vehicle_recalls_and_faults_alert",
+      get(:validatable_document_factories).vehicle_recalls_and_faults_alert_factory)
```
Comment (tommyp) [inline app/lib/specialist_publisher_wiring.rb]: As above.

File: README.md [hunk 4]
```diff
@@ -49,19 +49,19 @@ $ bundle exec rake
...
+  - `document_type_publication_alert_formatter.rb`
```
Comment (jamiecobbett) [inline README.md]: I know it's not something you've changed, but (assuming I'm correct - @tommyp?) could you append `for email-alert-api`?

Comment (tommyp) [inline README.md]: That's exactly what it's for. Good call.

File: app/lib/specialist_publisher_wiring.rb (hunk 2 above)
Comment (tadast) [inline app/lib/specialist_publisher_wiring.rb]: I just made it consistent with the rest of the similar method invocations in the file. I can update them all, which one do you prefer?

1
```
  define_factory(:cma_case_builder) {
    SpecialistDocumentBuilder.new("cma_case",
      get(:validatable_document_factories).cma_case_factory
    )
  }
```

2
```
  define_factory(:cma_case_builder) {
    SpecialistDocumentBuilder.new(
      "cma_case",
      get(:validatable_document_factories).cma_case_factory
    )
  }
```

3
```
  define_factory(:cma_case_builder) {
    SpecialistDocumentBuilder.new("cma_case",
                                  get(:validatable_document_factories).cma_case_factory
    )
  }
```

Comment (tommyp) [inline app/lib/specialist_publisher_wiring.rb]: If you're making it similar, it's OK to leave it then.

**Comment by tadast** [comment]
Comments addressed, anything else? :)

**Comment by tadast** [comment]
I've completed the checklist in the PR description and addressed all the comments. Can we merge this please?

**Pull request merged**